homeassistant/components/subaru/sensor.py

```diff
@@ -146,123 +153,96 @@ async def async_setup_entry(
-    coordinator = hass.data[DOMAIN][config_entry.entry_id][ENTRY_COORDINATOR]
-    vehicle_info = hass.data[DOMAIN][config_entry.entry_id][ENTRY_VEHICLES]
+    entry = hass.data[DOMAIN][config_entry.entry_id]
+    coordinator = entry[ENTRY_COORDINATOR]
+    vehicle_info = entry[ENTRY_VEHICLES]
     entities = []
-    for vin in vehicle_info:
-        entities.extend(create_vehicle_sensors(vehicle_info[vin], coordinator))
-    async_add_entities(entities, True)
+    for _, info in vehicle_info.items():
+        entities.extend(create_vehicle_sensors(info, coordinator))
+    async_add_entities(entities)
 
 
 def create_vehicle_sensors(vehicle_info, coordinator):
     """Instantiate all available sensors for the vehicle."""
-    sensors_to_add = []
+    sensor_descriptions_to_add = []
     if vehicle_info[VEHICLE_HAS_SAFETY_SERVICE]:
-        sensors_to_add.extend(SAFETY_SENSORS)
+        sensor_descriptions_to_add.extend(SAFETY_SENSORS)
 
         if vehicle_info[VEHICLE_API_GEN] == API_GEN_2:
-            sensors_to_add.extend(API_GEN_2_SENSORS)
+            sensor_descriptions_to_add.extend(API_GEN_2_SENSORS)
 
         if vehicle_info[VEHICLE_HAS_EV]:
-            sensors_to_add.extend(EV_SENSORS)
+            sensor_descriptions_to_add.extend(EV_SENSORS)
 
     return [
         SubaruSensor(
             vehicle_info,
             coordinator,
-            s[SENSOR_TYPE],
-            s[SENSOR_CLASS],
-            s[SENSOR_FIELD],
-            s[SENSOR_UNITS],
+            description,
         )
-        for s in sensors_to_add
+        for description in sensor_descriptions_to_add
     ]
 
 
-class SubaruSensor(SubaruEntity, SensorEntity):
+class SubaruSensor(CoordinatorEntity, SensorEntity):
     """Class for Subaru sensors."""
 
     def __init__(
-        self, vehicle_info, coordinator, entity_type, sensor_class, data_field, api_unit
+        self,
+        vehicle_info,
+        coordinator,
+        description,
     ):
         """Initialize the sensor."""
-        super().__init__(vehicle_info, coordinator)
-        self.hass_type = "sensor"
-        self.current_value = None
-        self.entity_type = entity_type
-        self.sensor_class = sensor_class
-        self.data_field = data_field
-        self.api_unit = api_unit
-
-    @property
-    def device_class(self):
-        """Return the class of this device, from component DEVICE_CLASSES."""
-        if self.sensor_class in DEVICE_CLASSES:
-            return self.sensor_class
-        return None
-
-    @property
-    def icon(self):
-        """Return the icon of the sensor."""
-        if not self.device_class:
-            return ICONS.get(self.entity_type)
-        return None
+        super().__init__(coordinator)
+        self.vin = vehicle_info[VEHICLE_VIN]
+        self.entity_description = description
+        self._attr_device_info = get_device_info(vehicle_info)
+        self._attr_name = f"{vehicle_info[VEHICLE_NAME]} {description.name}"
```

Comment:
See https://developers.home-assistant.io/docs/core/entity?_highlight=has_en#entity-naming

```suggestion
```

We'll probably need to update the test entity_ids, but existing entity_ids for the users will remain the same as they are registered in the entity registry.